org.eclipse.cdt.cross.arc.gnu/src/org/eclipse/cdt/cross/arc/gnu/ARCManagedCommandLineGenerator.java

```diff
@@ -390,6 +393,26 @@ else if ((oValue instanceof Boolean)) {
                             }
                         }
                     }
+                    
+                    if (oTool.getBaseId().contains("compiler") || oTool.getBaseId().contains("assembler")) {
+                        if (tcf_import_c_defines != null && tcf_import_c_defines) {
+                            try {
+                            	if (Files.exists(Paths.get(projectBuildPath)))
+                            	{
+	                                Files.write(Paths.get(projectBuildPath + File.separator + fileContent.getCDefinesFileName()), fileContent.getCDefinesText().getBytes(),
+	                                        StandardOpenOption.CREATE);
+                            	}
+                            } catch (IOException e1) {
+                                StatusManager.getManager().handle(
+                                        new Status(IStatus.ERROR, ARCPlugin.PLUGIN_ID, e1.getMessage()),
+                                        StatusManager.SHOW);
+                                e1.printStackTrace();
+                            }
+                            if (Files.exists(Paths.get(projectBuildPath))) {
+                                oList_gcc_options.add("-I " + projectBuildPath);
```

Comment:
Option `-I` specifies where to search for header files which are included via `#include` directive. Your header file is no included directly by the user code, therefore it should be included via command line option - you need an option that includes a file, rather then specifies directory for search.